lib/service/service.go

```diff
@@ -586,19 +642,23 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 		}
 
 		proxyLimiter.WrapHandle(webHandler)
-
 		process.BroadcastEvent(Event{Name: ProxyWebServerEvent, Payload: webHandler})
 
 		log.Infof("[PROXY] init TLS listeners")
-		err = utils.ListenAndServeTLS(
+		webListener, err = utils.ListenTLS(
 			cfg.Proxy.WebAddr.Addr,
-			proxyLimiter,
 			cfg.Proxy.TLSCert,
 			cfg.Proxy.TLSKey)
 		if err != nil {
 			return trace.Wrap(err)
 		}
-
+		if err = http.Serve(webListener, proxyLimiter); err != nil {
+			if askedToExit {
```

Comment:
see my comments above. when there's no a synchronized object (like a socket) I used `CloseBroadcaster`. There is simply no need to synchronize the process of closing a socket.